build/storage/scripts/disk_infrastructure.py

```diff
@@ -0,0 +1,178 @@
+import base64
+import json
+import logging
+import re
+import socket
+import subprocess
+import time
+import uuid
+
+from scripts import socket_functions
+
+logging.root.setLevel(logging.CRITICAL)
+
+
+def get_number_of_virtio_blk(socket: str) -> int:
+    cmd = 'lsblk --output "NAME,VENDOR,SUBSYSTEMS"'
+    out = socket_functions.send_command_over_unix_socket(
+        sock=socket, cmd=cmd, wait_for_secs=1
+    )
+    number_of_virtio_blk_devices = len(re.findall("block:virtio:pci", out))
+    return number_of_virtio_blk_devices
+
+
+def is_virtio_blk_attached(socket: str) -> int:
+    if get_number_of_virtio_blk(socket) == 0:
+        logging.error("virtio-blk is not found")
+        return 1
+    logging.info("virtio-blk is found")
+    return 0
+
+
+def is_virtio_blk_not_attached(socket: str) -> int:
+    if is_virtio_blk_attached(socket):
+        return 0
+    return 1
+
+
+def check_number_of_virtio_blk_devices(
+    vm_serial: str, expected_number_of_devices: int
+) -> int:
+    number_of_devices = get_number_of_virtio_blk(vm_serial)
+    if number_of_devices != expected_number_of_devices:
+        logging.error(
+            f"Required number of devices '{expected_number_of_devices}' does "
+            f"not equal to actual number of devices '{number_of_devices}'"
+        )
+        return 1
+    else:
+        logging.info(f"Number of attached virtio-blk devices is '{number_of_devices}'")
+        return 0
+
+
+def create_and_expose_subsystem_over_tcp(
+    ip_addr: str, nqn: str, port_to_expose: int, storage_target_port: int
+) -> None:
+    subprocess.run(
```

Comment:
if you switching to python, use `requests` don't call back into shell

Reply:
Thanks for your review. This code is still a work in progress, I left out many shell calls, but the code will be changed. Do you want requests to be used explicitly or urllib would be sufficient?

Reply:
I think requests is better, example:
```
import json
import requests
 
if __name__ == '__main__':
        payload = {'id':1, 'method': 'bdev_get_bdevs', 'params': {'name': 'Malloc0'}}
        url = 'http://10.10.10.1:8080/'
        req = requests.post(url,
                        data=json.dumps(payload),
                        auth=('user', 'password'),
                        verify=False,
                        timeout=30)
        print (req.json())
```

Reply:
I couldn't use requests module but I implemented another pythonic solution. I used client classes from rpc.py and sma-client.py to send requests.